test/Rx/Functional/Operator/IsEmptyTest.php

```diff
@@ -0,0 +1,170 @@
+     */
+    public function should_raise_error_if_source_raise_error()
+    {
+        $e = new \Exception();
+
+        $xs = $this->createHotObservable([
+            onError(300, $e),
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function() use ($xs) {
+            return $xs->isEmpty();
+        });
+
+        $this->assertMessages([
+            onError(300, $e),
+        ], $results->getMessages());
+
+        $this->assertSubscriptions([
+            subscribe(200, 300),
+        ], $xs->getSubscriptions());
+    }
+
+    /**
+     * @test
+     */
+    public function should_not_completes_if_source_never_emits()
+    {
+        $xs = $this->createHotObservable([]);
+
+        $results = $this->scheduler->startWithCreate(function() use ($xs) {
+            return $xs->isEmpty();
+        });
+
+        $this->assertMessages([], $results->getMessages());
+        $this->assertSubscriptions([
+            subscribe(200, 1000),
+        ], $xs->getSubscriptions());
+    }
+
+    /**
+     * @test
+     */
+    public function should_return_true_if_source_completes_immediately()
+    {
+        $xs = $this->createHotObservable([
+            onCompleted(201),
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function() use ($xs) {
+            return $xs->isEmpty();
+        });
+
+        $this->assertMessages([
+            onNext(201, true),
+            onCompleted(201),
+        ], $results->getMessages());
+        $this->assertSubscriptions([
+            subscribe(200, 201),
+        ], $xs->getSubscriptions());
+    }
+
+    /**
+     * @test
+     */
+    public function should_allow_unsubscribing_explicitly_and_early()
+    {
+        $xs = $this->createHotObservable([
+            onNext(600, 'a'),
+            onNext(700, 'b'),
+        ]);
+
+        $results = $this->scheduler->startWithDispose(function() use ($xs) {
+            return $xs->isEmpty();
+        }, 500);
+
+        $this->assertMessages([], $results->getMessages());
+
+        $this->assertSubscriptions([
+            subscribe(200, 500),
+        ], $xs->getSubscriptions());
+    }
+
+    /**
+     * @test
+     */
+    public function should_not_break_unsubscription_chains_when_result_is_unsubscribed_explicitly()
+    {
+        $xs = $this->createHotObservable([
+            onNext(600, 'a'),
+            onNext(700, 'b'),
+        ]);
+
+        $results = $this->scheduler->startWithDispose(function() use ($xs) {
+            return $xs
+                ->flatMap(function($value) {
+                    return \Rx\Observable::just($value);
+                })
+                ->isEmpty()
+                ->flatMap(function($value) {
+                    return \Rx\Observable::just($value);
+                });
+        }, 500);
+
+        $this->assertMessages([], $results->getMessages());
+
+        $this->assertSubscriptions([
+            subscribe(200, 500),
+        ], $xs->getSubscriptions());
+    }
+
+}
```

Comment:
This should be already covered by [should_not_completes_if_source_never_emits ](https://github.com/ReactiveX/RxPHP/pull/125/files/d78d3a3d84bce819ac0e936f8b73f8a1bc4b7574#diff-b99fb56076d5d32fbf2472d605dbf506R85) test.